numpy/core/src/multiarray/ctors.c

```diff
@@ -1893,26 +1897,15 @@ PyArray_FromInterface(PyObject *origin)
         return NULL;
     }
 
-    /* Allow unicode type strings */
-    if (PyUnicode_Check(attr)) {
-        PyObject *tmp = PyUnicode_AsASCIIString(attr);
-        if (tmp == NULL) {
-            goto fail;
-        }
-        attr = tmp;
-    }
-    else {
-        Py_INCREF(attr);
```

Comment:
This incref is now missing, we have to delete the line below (1931):
```
    Py_DECREF(attr);  /* Pairs with the unicode handling above */
```
(and possibly some error handling in between).

Reply:
That is why I moved ` tmp` to global so that it could be decrefed on error exit and cleared on block exit. Essentially the code "borrows" a reference from `tmp`. The original error handling needed fixing.

Reply:
I thought the way that this code did it before felt fairly idiomatic, and wasn't a huge fan of the approach in #17145. The code wasn't borrowing a reference from `tmp`, it was changing the semantics of `attr` to hold a new reference instead of a borrowed reference.

Either way, removed the line @seberg found, thanks for finding that.

Reply:
Thanks, seems like some bugs trying to creep out every time. A void test in the full test run is now failing with a SystemError:
```
a = np.empty((4,), dtype='V4')
a_view = as_strided(a, shape=(3, 4), strides=(0, a.itemsize))
```
The `np.lib.stride_tricks.as_strided` call somehow returns an error without setting an error value for whatever array-interface `a` has, I guess.

Reply:
Fixed I think...